File: android/lib/src/main/java/org/elastos/sdk/elephantwallet/contact/ContactStatus.java
package org.elastos.sdk.elephantwallet.contact;

public enum ContactStatus {
    Invalid(0x0),
    WaitForAccept(0x1),
    Offline(0x2),
    Online(0x4),
    Removed(0x8);

    public static ContactStatus valueOf(int id) {
        ContactStatus[] values = ContactStatus.values();
        for(int idx = 0; idx < values.length; idx++) {
            if(values[idx].id == id) {
                return values[idx];
            }
        }
        return null;
    }

    private ContactStatus(int id){
                       this.id = id;
                                    }
    private int id;
}

File: ios/sdk/swift/internal/ContactStatus.swift

public enum ContactStatus: Int, Codable {
  case Invalid = 0x0
  case WaitForAccept = 0x1
  case Offline = 0x2
  case Online = 0x4
  case Removed = 0x8
  
  public init(from decoder: Decoder) throws {
    let container = try decoder.singleValueContainer()
    let x = try container.decode(String.self)
    switch x {
    case "Invalid":
      self = .Invalid
      break
    case "WaitForAccept":
      self = .WaitForAccept
      break
    case "Offline":
      self = .Offline
      break
    case "Online":
      self = .Online
      break
    case "Removed":
      self = .Removed
      break
    default:
      throw DecodingError.typeMismatch(ContactStatus.self,
                                       DecodingError.Context(codingPath: decoder.codingPath,
                                                             debugDescription: "Wrong type for InnerItem"))
    }
  }

  public func encode(to encoder: Encoder) throws {
    var container = encoder.singleValueContainer()
    switch self {
    case .Invalid:
      try container.encode("Invalid")
      break
      case .WaitForAccept:
        try container.encode("WaitForAccept")
        break
      case .Offline:
        try container.encode("Offline")
        break
      case .Online:
        try container.encode("Online")
        break
      case .Removed:
        try container.encode("Removed")
        break
    }
  }
}
